src/utils/ExtrusionLine.cpp

```diff
@@ -27,4 +28,180 @@ void ExtrusionLine::appendJunctionsTo(LineJunctions& result) const
+     From this area we compute the height of the representative triangle using
+     the standard formula for a triangle area: A = .5*b*h
+     */
+    coord_t accumulated_area_removed = previous.p.X * current.p.Y - previous.p.Y * current.p.X; // Twice the Shoelace formula for area of polygon per line segment.
+
+    for (size_t point_idx = 1; point_idx < junctions.size() - 1; point_idx++)
+    {
+        current = junctions.at(point_idx);
+
+        // don't spill over if the [next] vertex will then be equal to [previous]
+        if (point_idx + 1 == junctions.size() && new_junctions.size() > 1)
+        {
+            next = new_junctions[0]; //Spill over to new polygon for checking removed area.
+        }
+        else
+        {
+            next = junctions.at(point_idx + 1);
+        }
+        const coord_t removed_area_next = current.p.X * next.p.Y - current.p.Y * next.p.X; // Twice the Shoelace formula for area of polygon per line segment.
+        const coord_t negative_area_closing = next.p.X * previous.p.Y - next.p.Y * previous.p.X; // area between the origin and the short-cutting segment
+        accumulated_area_removed += removed_area_next;
+
+        const coord_t length2 = vSize2(current - previous);
+        if (length2 < 25)
+        {
+            // We're allowed to always delete segments of less than 5 micron. The width in this case doesn't matter that
+            // much.
+            continue;
+        }
+
+        const coord_t area_removed_so_far = accumulated_area_removed + negative_area_closing; // close the shortcut area polygon
+        const coord_t base_length_2 = vSize2(next - previous);
+
+        if (base_length_2 == 0) //Two line segments form a line back and forth with no area.
+        {
+            continue; //Remove the vertex.
+        }
+        //We want to check if the height of the triangle formed by previous, current and next vertices is less than allowed_error_distance_squared.
+        //1/2 L = A           [actual area is half of the computed shoelace value] // Shoelace formula is .5*(...) , but we simplify the computation and take out the .5
+        //A = 1/2 * b * h     [triangle area formula]
+        //L = b * h           [apply above two and take out the 1/2]
+        //h = L / b           [divide by b]
+        //h^2 = (L / b)^2     [square it]
+        //h^2 = L^2 / b^2     [factor the divisor]
+        const coord_t height_2 = area_removed_so_far * area_removed_so_far / base_length_2;
+        coord_t weighted_average_width;
+        const coord_t extrusion_area_error = calculateExtrusionAreaDeviationError(previous, current, next, weighted_average_width);
+        if ((height_2 <= 1 //Almost exactly colinear (barring rounding errors).
+             && LinearAlg2D::getDistFromLine(current.p, previous.p, next.p) <= 1) // make sure that height_2 is not small because of cancellation of positive and negative areas
```

Comment:
```suggestion
             && LinearAlg2D::getDistFromLine(current.p, previous.p, next.p) <= 1) // Make sure that height_2 is not small because of cancellation of positive and negative areas
```